### ISSUE TITLE
Add extendable renewable carriers to `extendable_carriers'/`Generator`? 

### ISSUE THREAD

**Issue by FabianHofmann** [PyPSA/pypsa-eur#355]
This can allow the flexibility to run an optimization with or without expansion or renewable technologies.

**Comment by euronion** [comment]
Do you mean something like the `expansion_limit` configuration option we added here: https://github.com/PyPSA/pypsa-eur/pull/352 ?

**Comment by FabianHofmann** [comment]
Yes, but I think both can/should exist in parallel. Assume you don't want to build out onshore wind but only use capacities from today and the other renewables could be extended by 10%. Then, you'd remove `onwind` from `extendable_carriers` and set the expansion limit.

**Comment by euronion** [comment]
Gotcha!

@fneum and I were also discussing yesterday about being able to provide a dictionary of countries + expansion_limit for each country. One could Combine, such that one can supply a number (all carriers + countries), a dict of carriers, a dict of countries or a dict of countries and carriers.
